gateway-ha/src/main/java/io/trino/gateway/baseapp/AppModule.java

```diff
@@ -46,11 +46,19 @@ protected int getApplicationPort()
                         .map(SimpleServerFactory::getConnector);
 
         return connectors
-                .filter(connector -> connector.getClass().isAssignableFrom(HttpConnectorFactory.class))
-                .map(connector -> (HttpConnectorFactory) connector)
-                .mapToInt(HttpConnectorFactory::getPort)
-                .findFirst()
-                .orElseThrow(IllegalStateException::new);
+              .filter(connector -> connector instanceof HttpConnectorFactory || connector instanceof HttpsConnectorFactory)
+              .map(connector -> {
+                  if (connector instanceof HttpConnectorFactory) {
+                      return ((HttpConnectorFactory) connector).getPort();
+                  } else if (connector instanceof HttpsConnectorFactory) {
+                      return ((HttpsConnectorFactory) connector).getPort();
+                  } else {
+                      throw new IllegalStateException("Unknown connector type");
+                  }
+              })
+              .mapToInt(Integer::intValue)
+              .findFirst()
+              .orElseThrow(IllegalStateException::new);
```

Comment:
Please add a test.

Reply:
Just to clarify you are saying to write a test where if the protocol specified is HTTPS then it should fail with an IllegalStateException?

Reply:
The test for the entire PR change.